### PR TITLE
Automatic Client Test

### PR DESCRIPTION
**Pull request by ejohnstown**
1. Add an SFTP automated test.
2. Update wolfSSH_SFTP_STAT() with clearing the error register if it is WANT_READ.

### CODE REVIEW

File: examples/sftpclient/sftpclient.c [hunk 1]
```diff
@@ -283,6 +285,7 @@ static void ShowUsage(void)
     printf(" -N            use non blocking sockets\n");
     printf(" -e            use ECC user authentication\n");
     /*printf(" -E            use ECC server authentication\n");*/
+    printf(" -A            automated test\n");
```
[the remark this answers is not in the document]
Comment (ejohnstown) [reply examples/sftpclient/sftpclient.c]: Changing it to something else. It's going to be like SCP implemented in SFTP.

File: examples/sftpclient/sftpclient.c [hunk 2]
```diff
@@ -44,6 +41,16 @@ static char* workingDir;
 #define MAX_CMD_SZ 7
 
 
+#ifndef WS_MAX_AUTOTEST_COUNT
```
Comment (JacobBarthelmeh) [inline examples/sftpclient/sftpclient.c]: possibly left over from previous commit

Comment (ejohnstown) [reply examples/sftpclient/sftpclient.c]: Oops.

File: examples/sftpclient/sftpclient.c [hunk 3]
```diff
@@ -1142,6 +1148,48 @@ int doCmds(func_args* args)
...
+            ret = wolfSSH_SFTP_Put(ssh, local, fullpath, 0, NULL);
+        }
+        else if (cmd == AUTOPILOT_GET) {
+            ret = wolfSSH_SFTP_Get(ssh, fullpath, local, 0, NULL);
+        }
+        err = wolfSSH_get_error(ssh);
+    } while ((err == WS_WANT_READ || err == WS_WANT_WRITE) &&
+            ret != WS_SUCCESS);
+
+    if (cmd == AUTOPILOT_PUT) {
```
Comment (JacobBarthelmeh) [inline examples/sftpclient/sftpclient.c]: this always prints now. We need a check for failure here

**Pull request merged**